OpenRA.Game/MusicPlaylist.cs

```diff
@@ -0,0 +1,105 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Traits;
+
+namespace OpenRA
+{
+	public class MusicPlaylist
+	{
+		bool installed = false;
+		MusicInfo currentSong = null;
+		MusicInfo[] music;
+		MusicInfo[] random;
+
+		public MusicInfo[] GetPlaylist { get { return music; } }
+		public bool GetInstalled { get { return installed; } }
+
+		public MusicPlaylist(World world)
+		{
+			installed = world.Map.Rules.InstalledMusic.Any();
+
+			music = world.Map.Rules.InstalledMusic.Select(a => a.Value).ToArray();
+			if (!installed)
+				return;
+
+			random = music.Shuffle(Game.CosmeticRandom).ToArray();
+
+			currentSong = Game.Settings.Sound.Shuffle ? random.First() : music.First();
+
+			Play();
```

Comment:
MapMusic is the ShellMap / Mission Music checkbox in the UI and even the current bleed tracks silenced music. If I want to disable music overall then the slider is the option.